### ISSUE TITLE
Make the new variable mu4e-compose-crypto-policy context dependent

### ISSUE THREAD

**Issue by nils-schween** [djcb/mu#1730]
Hi everyone,

it's cool to see that mu is progressing so fast ultimately and I am really happy to see that there are also some changes related to the crypto setup.
Yesterday I tested the new variable `mu4e-compose-crypto-policy`, which was included int the pull-request https://github.com/djcb/mu/pull/1610, and it works almost perfectly. 
But I noticed that I was not able to restrict its settings to a specific context. 

## Expected or desired behavior
I expected that if I put it into the `:vars` section of the `make-mu4e-context` function to work. But it did not. Here is what I did :
```elisp
:vars '( ( user-mail-address       . "my-mail@address.com"  )
		       ( user-full-name          . "My Name" )
		       ( mu4e-sent-folder        . "/path/to/Sent" )
		       ( mu4e-drafts-folder      . "/path/to/Drafts")
		       ( mu4e-trash-folder       . "/path/to/Trash")
		       ( smtpmail-smtp-server . "mailserver.com")
		       ( smtpmail-smtp-service . 25)
		       ( mu4e-compose-signature-auto-include . t)
                       
                       (mu4e-compose-crypto-policy '(sign-all-messages encrypt-encrypted-replies))
		       
                       ( mu4e-compose-signature .
						(concat
						 "My signature line 1\n"
                                                 "line2\n"
						 )))
```
## Actual behavior
But it works if I set the variable outside the `make-mu4e-context`, i.e. including the following line in .emacs file
```elisp 
(setq mu4e-compose-crypto-policy '(sign-all-messages encrypt-encrypted-replies))
```
Unfortunately, it is then activated for all my contexts.

## Versions of mu, mu4e/emacs, operating system etc.
I am using the current version of mu, i.e. 1.5.3. and the following GNU Emacs 26.3 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20) of 2020-05-19.

## Any other detail
There is one more thing I noticed: When you activate signing or encrypting your mails, then a line like this 
```
<#secure method=pgpmime mode=sign>
```
is inserted at the beginning of your compose buffer. After having filled the to and the subject field, I jump to the text part of the buffer using `C-c C-b`. Since there is this new line, the point jumps directly in front of it and I cannot directly start typing. It would be great if `C-c C-b` could overjump this new line. 

Thanks a lot,
Nils

**Comment by djcb** [comment]
Hmm, seems to work fine for me. In your example, seems you're missing the `.` between variable name and value... Might that be the problem?

**Comment by nils-schween** [comment]
@djcb you are right, I forgot to put a `.` 
The correct line looks as follows:
```elisp
(mu4e-compose-crypto-policy . (sign-all-messages encrypt-encrypted-replies))
```
Thanks a lot! And I am sorry for having overlooked the `.` 
Nils

**Comment by nils-schween** [comment]
Hi,
I am sorry for reopening this issue. But it's still not working as expected: I have different gpg keys for different contexts and I expected that the key corresponding to the email address of the context I am currently using would be used to sign and encrypt emails. For example, I am in context "A" and in context "A" I have made the following setting:
```elisp
(mu4e-compose-crypto-policy . (sign-all-messages encrypt-encrypted-replies))
```
When I enter the `mu4e:compose` mode, it looks fine: there is the line `<#secure method=pgpmime mode=sign>`, but on sending the e-mail I am asked to enter a password for a key corresponding to an email which belongs to a different context, say "B".
Thus, I am using the wrong key to sign my email. I expected, that the key is chosen depending on the context I am currently in.

And I also noticed another but minor problem: When setting `mu4e-compose-crypto-policy` for one context, it's values will be used in all other contexts. Thus I was required to make the following setting in all contexts, where I did not want to use any crypto setup:
```elisp
(mu4e-compose-crypto-policy . nil)
```
I did not expect that setting the variable in one context, would change it's value for other contexts. 

Once more: sorry for reopening. Help is appreciated.
Nils

**Comment by nils-schween** [comment]
I could solve the first problem by using the following mml setting inside my .emacs file:
```elisp
(setq  mml-secure-openpgp-sign-with-sender t)
```
When the documentation is updated to include the new variable `mu4e-compose-crypto-policy`, it would make sense to explicitly note that it is necessary to set the above variable to make it work in different contexts. It is mentioned in the FAQ: "_C.3.11 How can I read encrypted messages that I sent?_" But I do not know if the snippet in the FAQ is still up to date. I tried to check the Emacs MIME manual, but the `mml-opengp-variables` weren't in there (https://www.gnu.org/software/emacs/manual/html_mono/emacs-mime.html), but  I found them in the file mml-sec.el file. All the mml2015 variables are declared obsolete since emacs 25.1 and the mml2015-use function does not exist anymore.  Maybe a modern form of the snippet may look like this
```elisp
;; GnuPG ID of your default identity.
(setq epg-user-id "key-id") ; NOTE: This is necessary to determine the default key for *file* encryption 

(setq mml-secure-openpgp-encrypt-to-self t   ;; enable that i can read the messages i encrypted with the the public key of someone else 
      mml-secure-openpgp-sign-with-sender t
      )

The second issue ( setting the crypto-policy in context, changes the crypto-policy in the other contexts if they are nil) remains.
